File: backend/SkillProof/SkillProof.Entities/Dtos/Tests/CandidateQuestionDto.cs
using SkillProof.Entities.Enums;

namespace SkillProof.Entities.Dtos.Tests;

public class CandidateQuestionDto
{
    public string Id { get; set; } = string.Empty;
    public QuestionType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string QuestionText { get; set; } = string.Empty;
    public string Language { get; set; } = string.Empty;
    public DifficultyLevel Difficulty { get; set; }

    public CandidateMultipleChoicePayloadDto? MultipleChoice { get; set; }
    public CandidateCodeCompletionPayloadDto? CodeCompletion { get; set; }
    public CandidateFillInTheBlankPayloadDto? FillInTheBlank { get; set; }
    public CandidateTrueFalsePayloadDto? TrueFalse { get; set; }
}

File: web/src/app/Models/Dtos/Test/candidate-question-dto.ts
import { DifficultyLevel } from '../../Enums/DifficultyLevel';
import { QuestionType } from '../../Enums/QuestionType';
import { CandidateCodeCompletionPayloadDto } from './candidate-code-completion-payload-dto';
import { CandidateFillInTheBlankPayloadDto } from './candidate-fill-in-the-blank-payload-dto';
import { CandidateMultipleChoicePayloadDto } from './candidate-multiple-choice-payload-dto';
import { CandidateTrueFalsePayloadDto } from './candidate-true-false-payload-dto';

export interface CandidateQuestionDto {
  id: string;
  type: QuestionType;
  title: string;
  questionText: string;
  language: string;
  difficulty: DifficultyLevel;

  multipleChoice?: CandidateMultipleChoicePayloadDto;
  codeCompletion?: CandidateCodeCompletionPayloadDto;
  fillInTheBlank?: CandidateFillInTheBlankPayloadDto;
  trueFalse?: CandidateTrueFalsePayloadDto;
}
